crates/ruff/src/checkers/ast.rs

```diff
@@ -4364,14 +4363,131 @@ impl<'a> Checker<'a> {
         for scope_index in self.scope_stack.iter().rev() {
             let scope = &self.scopes[*scope_index];
 
-            if matches!(scope.kind, ScopeKind::Class(_)) {
+            if let ScopeKind::Class(def) = &scope.kind {
                 if id == "__class__" {
                     return;
                 } else if !first_iter && !in_generator {
                     continue;
                 }
+                for base in def.bases.iter() {
```

Comment:
Could we instead colocate this with the logic that ensures we treat these as runtime annotations when futures aren't enabled? I.e., here:

```rust
StmtKind::AnnAssign {
    target,
    annotation,
    value,
    ..
} => {
    // If we're in a class or module scope, then the annotation needs to be
    // available at runtime.
    // See: https://docs.python.org/3/reference/simple_stmts.html#annotated-assignment-statements
    if !self.annotations_future_enabled
        && matches!(
            self.current_scope().kind,
            ScopeKind::Class(..) | ScopeKind::Module
        )
    {
        ...
```